typescript/web/src/components/datasets/upsert-dataset-modal.tsx

```diff
@@ -136,7 +136,7 @@ export const UpsertDatasetModal = ({
 
   useEffect(() => {
     if (
-      existingDataset != null &&
+      existingDataset?.searchDataset != null &&
```

Comment:
Maybe we should check here that `existingDataset?.searchDataset?.id != null` as it is the property that we are comparing to the current datased ID later on?

Reply:
```suggestion
      existingDataset?.searchDataset?.id != null &&
```